core/src/main/scala/org/http4s/parser/OriginHeader.scala

```diff
@@ -9,13 +9,20 @@ package parser
 
 import cats.data.NonEmptyList
 import java.nio.charset.{Charset, StandardCharsets}
+
 import org.http4s._
 import org.http4s.headers.Origin
 import org.http4s.internal.parboiled2._
+import cats.parse.{Parser, Parser1, Rfc5234}
+import org.http4s.Uri.RegName
+import cats.implicits._
 
 trait OriginHeader {
-  def ORIGIN(value: String): ParseResult[Origin] =
-    new OriginParser(value).parse
+  def ORIGIN(value: String): ParseResult[Origin] = {
```

Comment:
Most of these are moving directly to the objects.  I'd move this to `Origin.parse`.  And if nothing else references the old parboiled parser, we should be able to delete this whole file. :tada:

Reply:
Looks like it is being used in `HttpHeaderParser` which is used in a bunch of places. Should I go through the process of removing it from `HttpHeaderParser`? Or leave it as is?

Reply:
If it can be ripped out easily, great.  If not, let's bring this in for a landing and go further in another PR.